### DISCUSSION TITLE
Static sites and rewrite urls

### DISCUSSION THREAD

**Discussion by codenutt** [vercel/community#2690] [Help]
### Summary

Would using a root site and rewrite URLs like described here:
https://vercel.com/guides/how-can-i-serve-multiple-projects-under-a-single-domain

Stop us from using static deployed sites or would everything still function like a static one?

### Steps to Reproduce

Having a root site with just a vercel.json file with rewrite rules in it. Those rewrite rules would then point to three statically built sites. In that case, would everything still be a static site?

**Comment by smaeda-ks** [comment]
Hi @codenutt, could you elaborate on your end goal more? What's the exact use case?
Also, what do you mean by "would everything still be a static site" ?

If you're simply rewriting all requests to an external endpoint, it's basically the same as you're introducing reverse proxy. And external rewrites responses are not cached by Vercel CDN, so requests would always hit your external origin.

**Comment by z4nr34l** [comment]
Rewrite will serve content `as is` on remote origin. If your remote app will use SSG/SSR/ISR you would get static content with rewrite. So everything should work as it have to :)

If that helped, please consider marking my comment as answer so other would be able to find it faster.
